src/main/drivers/castle_telemetry_decode.c

```diff
@@ -0,0 +1,206 @@
+#include "platform.h"
+
+#ifdef USE_TELEMETRY_CASTLE
+
+#include "build/atomic.h"
+
+#include "drivers/castle_telemetry_decode.h"
+#include "drivers/nvic.h"
+#include "drivers/pwm_output.h"
+#include "drivers/timer.h"
+
+typedef struct castleInterrupt_s {
+    timerCCHandlerRec_t pwmEdgeCb;
+    volatile timCCR_t *timingChannelCCR;
+    volatile timCCR_t *directChannelCCR;
+    timerChannel_t* timer;
+    timCCR_t outputEnableTime;
+    timCCR_t pwmEdge;
+    timCCR_t saveCCR;
+    castleTelemetry_t telem[2];
+    uint8_t directChannelIndex;
+    uint8_t whichTelem; // telem we are writing to (0 or 1).
+    uint8_t telemIndex; // where in the telem struct are we?
+} castleInterrupt_t;
+
+static FAST_DATA_ZERO_INIT castleInterrupt_t castleState;
+
+void getCastleTelemetry(castleTelemetry_t* telem) {
```

Comment:
Format. { is on a new line.

Reply:
Fixed.  I noticed all the formatting issues and realized I hadn't run astyle with the right options, so I re-ran that on this file and the new part of esc_sensor.c